### PR TITLE
Add Markdown Support

### PR DESCRIPTION
**Pull request by hlavu**
Hi Quan, I have completed the code for the markdown support. 
Here are some changes I had made:

* In `mini-ssg.js`:
  - Modify the file handling so it will accept the ".md" extension. 
  
* In `generateHTML.js`:
  - Render Heading 1.
  - Render Heading 2.
  - Render Link.
 
* In `README.md`:
   - Update README.md to add documentation for Markdown support.

Please review it and let me know If you want me to change any part.

### CODE REVIEW

File: utils/generateHTML.js [hunk 1]
```diff
@@ -1,58 +1,77 @@
 const { readFile, writeFile } = require("fs/promises");
 const path = require("path");
 const createHtml = require("create-html");
+var htmlBody = "";
```
Comment (mqnguyen5) [inline utils/generateHTML.js]: You should change this line into `let`, as using `var` might cause some unexpected behaviour. A better explanation about why can be found [here](https://stackoverflow.com/questions/762011/whats-the-difference-between-using-let-and-var).

File: utils/generateHTML.js [hunk 2]
```diff
@@ -1,58 +1,77 @@
...
+                })
+                .join(" ")
+            );
+    } else {
+      //  handles ".md" content
+      htmlBody = data
+        .replace(/\[(.*?)\]\((.*?)\)/gim, "<a href='$2'>$1</a>") // replaces link -> <a href="$2">$1</a>
+        .split(/\r?\n\r?\n/)
+        .map(function (para) {
+          if (para.match(/(?<!#)#{1}\s/)) {
```
Comment (mqnguyen5) [inline utils/generateHTML.js]: From the Mozilla [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match):
> `String.match()` returns an Array whose contents depend on the presence or absence of the global (g) flag, or **null** if no matches are found. 

Thus, I suggest changing both the `if` and `else if` conditions to:
```js
if ([statement] != null) 
```

**Review by mqnguyen5 — CHANGES_REQUESTED** [review]
Overall, the pull request looks great. However, there are some things that should be changed and I have left comments so you can have a look.

**Comment by hlavu** [comment]
That's great, I will change them.

**Review by mqnguyen5 — COMMENTED** [review]
Everything looks great! Thanks for the contribution.

**Pull request merged**